src/v/cluster/partition_manager.cc

```diff
@@ -115,26 +115,52 @@ ss::future<consensus_ptr> partition_manager::manage(
   raft::keep_snapshotted_log keep_snapshotted_log) {
     gate_guard guard(_gate);
     auto dl_result = co_await maybe_download_log(ntp_cfg, rtp);
-    auto [logs_recovered, min_kafka_offset, max_kafka_offset, manifest]
+    auto
+      [logs_recovered,
+       clean_download,
+       min_offset,
+       max_offset,
+       manifest,
+       ot_state]
       = dl_result;
     if (logs_recovered) {
         vlog(
           clusterlog.info,
           "Log download complete, ntp: {}, rev: {}, "
-          "min_kafka_offset: {}, max_kafka_offset: {}",
+          "min_offset: {}, max_offset: {}",
           ntp_cfg.ntp(),
           ntp_cfg.get_revision(),
-          min_kafka_offset,
-          max_kafka_offset);
+          min_offset,
+          max_offset);
+        if (!clean_download) {
+            vlog(
+              clusterlog.warn,
+              "{} partition recovery is not clean, try to delete the topic "
+              "and retry recovery",
+              ntp_cfg.ntp());
+            manifest.disable_permanently();
```

Comment:
Also, it seems like this may still end up being different between replicas in the case of transient errors, no? Could we end up enabling the archival_meta_stm on some replicas but not on others? Perhaps we should update the segment download to not have a timeout and only accept 404s as real errors (though even that runs the risk of potentially racing with a malicious segment deletion)